http-client-spi/src/main/java/software/amazon/awssdk/http/HttpMetrics.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.http;
+
+import software.amazon.awssdk.metrics.MetricCategory;
+import software.amazon.awssdk.metrics.SdkMetric;
+
+/**
+ * Metrics collected by HTTP clients.
+ */
+public final class HttpMetrics {
+    /**
+     * The name of the HTTP client.
+     */
+    public static final SdkMetric<String> HTTP_CLIENT_NAME = metric("HttpClientName", String.class);
+
+    /**
+     * The maximum number of connections that will be pooled by the HTTP client.
+     */
+    public static final SdkMetric<Integer> MAX_CONNECTIONS = metric("MaxConnections", Integer.class);
```

Comment:
Do we need to namespace all sdk metrics?

Reply:
I think the issue with that is if you want to "pretty print" the metrics, or write your own publisher, there's no easy way to just get the metric name